packages/dds/merge-tree/src/localReference.ts

```diff
@@ -42,7 +42,7 @@ export function _validateReferenceType(refType: ReferenceType) {
  * @sealed
  */
 export interface LocalReferencePosition extends ReferencePosition {
-    callbacks?: Partial<Record<"beforeSlide" | "afterSlide", () => void>>;
+    callbacks?: Partial<Record<"beforeSlide" | "afterSlide", (ref: LocalReferencePosition) => void>>;
```

Comment:
adding ref to the callback, allows us to reuse a function, rather than needing a distinct function per reference, so less memory